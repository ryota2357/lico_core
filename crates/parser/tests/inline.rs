//! This file is generated by `cargo codegen parser`, do not edit by hand.
#![cfg_attr(rustfmt, rustfmt::skip)]

mod utils;
const GROUP: &str = "inline";

#[test]
fn expr_literals() {
    let source = include_str!("../test_data/inline/expr_literals.lico");
    let snapshot = utils::make_snapshot("expr_literals", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn param_list() {
    let source = include_str!("../test_data/inline/param_list.lico");
    let snapshot = utils::make_snapshot("param_list", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn attr_stmt() {
    let source = include_str!("../test_data/inline/attr_stmt.lico");
    let snapshot = utils::make_snapshot("attr_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn break_stmt() {
    let source = include_str!("../test_data/inline/break_stmt.lico");
    let snapshot = utils::make_snapshot("break_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn continue_stmt() {
    let source = include_str!("../test_data/inline/continue_stmt.lico");
    let snapshot = utils::make_snapshot("continue_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn func() {
    let source = include_str!("../test_data/inline/func.lico");
    let snapshot = utils::make_snapshot("func", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn return_stmt() {
    let source = include_str!("../test_data/inline/return_stmt.lico");
    let snapshot = utils::make_snapshot("return_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn var_stmt() {
    let source = include_str!("../test_data/inline/var_stmt.lico");
    let snapshot = utils::make_snapshot("var_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
#[test]
fn while_stmt() {
    let source = include_str!("../test_data/inline/while_stmt.lico");
    let snapshot = utils::make_snapshot("while_stmt", source);
    utils::assert_snapshot!(GROUP, snapshot);
}
