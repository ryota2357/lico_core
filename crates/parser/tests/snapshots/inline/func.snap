---
source: crates/parser/tests/inline.rs
---
# func

## Input

```
func foo() end
func() end
```

## CST

```
SOURCE_FILE@0..25
  FUNC_STMT@0..14
    FUNC_KW@0..4 "func"
    WHITESPACE@4..5 " "
    NAME@5..8
      IDENT@5..8 "foo"
    PARAM_LIST@8..10
      OPEN_PAREN@8..9 "("
      CLOSE_PAREN@9..10 ")"
    WHITESPACE@10..11 " "
    END_KW@11..14 "end"
  WHITESPACE@14..15 "\n"
  FUNC_EXPR@15..25
    FUNC_KW@15..19 "func"
    PARAM_LIST@19..21
      OPEN_PAREN@19..20 "("
      CLOSE_PAREN@20..21 ")"
    WHITESPACE@21..22 " "
    END_KW@22..25 "end"
```

## AST

```
SourceFile {
    shebang: none,
    stmt: [
        FuncStmt {
            func_token: FUNC_KW@0..4 "func",
            name: Name {
                ident: IDENT@5..8 "foo",
            },
            param_list: ParamList {
                open_paren_token: OPEN_PAREN@8..9 "(",
                params: [],
                close_paren_token: CLOSE_PAREN@9..10 ")",
            },
            statements: [],
            end_token: END_KW@11..14 "end",
        },
        FuncExpr {
            func_token: FUNC_KW@15..19 "func",
            param_list: ParamList {
                open_paren_token: OPEN_PAREN@19..20 "(",
                params: [],
                close_paren_token: CLOSE_PAREN@20..21 ")",
            },
            statements: [],
            end_token: END_KW@22..25 "end",
        },
    ],
}
```

vim:ft=markdown
